numba/targets/arraymath.py

```diff
@@ -614,6 +614,12 @@ def check_array(a):
         raise ValueError('zero-size array to reduction operation not possible')
 
 
+@register_jitable
+def _check_is_integer(v, name):
+    if not isinstance(v, (int, types.Integer)):
+        raise TypeError('{} must be an integer'.format(name))
```

Comment:
We generally tend to use `TypingError` in these cases rather than `TypeError`. This is to indicate that Numba failed during type inference.